src/lava/magma/compiler/builder.py

```diff
@@ -253,22 +284,18 @@ def set_csp_ports(self, csp_ports: ty.List[AbstractCspPort]):
             new_ports.setdefault(p.name, []).extend(
                 p if isinstance(p, list) else [p]
             )
-        self._check_not_assigned_yet(
-            self.csp_ports, new_ports.keys(), "csp_ports"
-        )
+
         # Check that there's a PyPort for each new CspPort
         proc_name = self.proc_model.implements_process.__name__
         for port_name in new_ports:
             if not hasattr(self.proc_model, port_name):
-                raise AssertionError(
-                    "PyProcessModel '{}' has \
-                no port named '{}'.".format(
-                        proc_name, port_name
-                    )
-                )
-        # Set new CspPorts
-        for key, ports in new_ports.items():
-            self.csp_ports.setdefault(key, []).extend(ports)
+                raise AssertionError("PyProcessModel '{}' has \
+                no port named '{}'.".format(proc_name, port_name))
```

Comment:
That indentation seems to be wrong. Be sure to target 100% linter score.

Reply:
PyCharm and flake8 do not give a warning on this, but I added the indent for consistency.